File: TypeScript/src/math/P9PalindromeNumber.ts
// Time: O(log n), Space: O(log n) // Approach: String reversal comparison
export function isPalindrome(x: number): boolean {
  if (x < 0) {
    return false;
  }
  const str = x.toString();
  const revertStr = str.split("").reverse().join("");
  return str === revertStr;
}

File: Java/src/main/java/org/adler/math/P9PalindromeNumber.java
package org.adler.math;

// Time: O(log n), Space: O(log n) // Approach: String reversal comparison
public class P9PalindromeNumber {
    public static boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        String tmpStr = String.valueOf(x);
        String reversed = new StringBuilder(tmpStr).reverse().toString();

        return tmpStr.equals(reversed);
    }
}
